tests/test_config.py

```diff
@@ -55,14 +54,12 @@ def test_hf_config_from_json(tmp_path):
     """Test for backward compatibility with older configs that didn't have the `hf_config` field."""
     from lit_gpt import Config
 
-    legacy_config = {
-        "name": "falcon-40b",
-        "org": "tiiuae",
-    }
+    legacy_config = {"name": "falcon-40b", "org": "tiiuae"}
     with open(tmp_path / "config.json", "w") as file:
         json.dump(legacy_config, file)
     new_config = Config.from_json(tmp_path / "config.json")
     assert new_config.name == "falcon-40b"
+    assert not hasattr(new_config, "org")
```

Comment:
What is the `deepcopy` fixing exactly? This assertion seems to pass in the main branch

Reply:
> What is the `deepcopy` fixing exactly? 

I guess I should have been more explicit 🤔.
For convenience, these lines copy already added config and extend it with additional versions:
https://github.com/Lightning-AI/lit-gpt/blob/069f6584a2a99609e288c32cb49c028bcf35b747/lit_gpt/config.py#L246-L250

The problem is that now we have a dictionary `hf_config` inside each config dictionary. 
https://github.com/Lightning-AI/lit-gpt/blob/069f6584a2a99609e288c32cb49c028bcf35b747/lit_gpt/config.py#L170-L173

So when we do a shallow copy and then change the name inside `hf_config` dictionary we change not the copy, but the value by the link. 
This results in an incorrect name inside config:
<img width="821" alt="Screenshot 2023-10-30 at 8 10 15 PM" src="https://github.com/Lightning-AI/lit-gpt/assets/58434077/b0d08829-6e55-49d3-b9f8-a6386a24a637">

Reply:
> This assertion seems to pass in the main branch

It should pass.
With current version of config the `org` key should be inside `hf_config`. For backward compatibility, the code automatically moves it to where it belongs:
https://github.com/Lightning-AI/lit-gpt/blob/069f6584a2a99609e288c32cb49c028bcf35b747/lit_gpt/config.py#L110-L113

So here I check that `org` key isn't at the “legacy” place.

Reply:
Could you add an assertion that fails in current `main` and passes with this PR checked out? You can use the example you used in your previous message. Thank you

Reply:
😄 
I initially [added a test](https://github.com/Lightning-AI/lit-gpt/pull/681/commits/81290ac6ac2789ba3f95fd12456c18935c0881e0) that can check this issue and cases when a short and hf names are different. 
But during the discussion with Adrian the decision was made to remove those tests.

The question is: should I return those tests, or should I do as you suggested?

Reply:
I believe Adrian wanted to keep the original test with different names, but it is fine to **add** a new parametrized test for all configs as you did in that commit

Reply:
Done.